File: app/src/main/java/it/gruppoinfor/home2work/models/Company.java
package it.gruppoinfor.home2work.models;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.android.gms.maps.model.LatLng;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Company  {

    @SerializedName("id")
    @Expose
    private Long id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("location")
    @Expose
    private LatLng location;
    @SerializedName("address")
    @Expose
    private Address address;
    @SerializedName("website")
    @Expose
    private String website;
    @SerializedName("karma")
    @Expose
    private Karma karma;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public LatLng getLocation() {
        return location;
    }

    public void setLocation(LatLng location) {
        this.location = location;
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    public String getWebsite() {
        return website;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    public Karma getKarma() {
        return karma;
    }

    public void setKarma(Karma karma) {
        this.karma = karma;
    }

    @Override
    public String toString() {
        return name + " (" + address.getCity() + ")";
    }
}

File: API/src/main/java/it/gruppoinfor/home2workapi/company/Company.kt
package it.gruppoinfor.home2workapi.company

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import it.gruppoinfor.home2workapi.FullAddress
import it.gruppoinfor.home2workapi.LatLng

import java.io.Serializable

class Company : Serializable {

    @SerializedName("Id")
    @Expose
    var id: Long = 0
    @SerializedName("Name")
    @Expose
    var name: String = ""
    @SerializedName("LatLng")
    @Expose
    var location: LatLng = LatLng()
    @SerializedName("Address")
    @Expose
    var address: FullAddress = FullAddress()

    override fun toString(): String {
        return "$name (${address.city})"
    }
}